hrms/hr/doctype/leave_block_list/leave_block_list.py

```diff
@@ -7,37 +7,56 @@
 import frappe
 from frappe import _
 from frappe.model.document import Document
+from frappe.utils import getdate
 
 
 class LeaveBlockList(Document):
 	def validate(self):
 		dates = []
 		for d in self.get("leave_block_list_dates"):
-
 			# date is not repeated
 			if d.block_date in dates:
 				frappe.msgprint(_("Date is repeated") + ":" + d.block_date, raise_exception=1)
 			dates.append(d.block_date)
 
+	@frappe.whitelist()
+	def get_weekly_off_dates(self, start_date, end_date, days, reason):
```

Comment:
Change the name to `set_weekly_off_dates` since it's not returning the dates but adding them to the table instead.

Reply:
Changed